libphal/phal_pdbg.C

```diff
@@ -280,4 +280,31 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc)
 	return validAttn;
 }
 
+bool hasControlTransitionedToPHYP()
+{
+  // If unable to read the register, by default consider the control is in hostboot
+  bool transitionedToPHYP = false;
+  //Read the scratch register to find if the host is transitioned to phyp
+  pdbg_target* pibTarget = pdbg_target_from_path(nullptr, "/proc0/pib");
+
+  uint64_t l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
+  // HB changes the below core scratch reg as one of the last instructions that is run,
+  // so if that is zero then we're in phyp
+  uint64_t l_coreScratchRegAddr = 0x4602F489;
+
+  // Is there any error in reading the scom address, assign default value to data
+  if (pib_read(pibTarget, l_coreScratchRegAddr, &l_coreScratchRegData))
+  {
+      log(level::ERROR, "scom read error: 0x%X", l_coreScratchRegAddr );
+      l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
```

Comment:
The assumption is returning false if the scratch read failed? then please add that to the documentation of the function.
Please return here if the read is failed.

Reply:
Agreed with @dhruvibm. If the scratch register reading fails then either we should be throwing an exception or returning false (if throwing an exception isn't an option)